File: Day02_OC加强_内存管理2/01-autoreleasepool的基本使用补充/03-学生类练习/Student.m
//
//  Student.m
//  01-autoreleasepool的基本使用补充
//
//  Created by SteveChiang on 15/12/21.
//  Copyright © 2015年 Apple.Inc. All rights reserved.
//

#import "Student.h"

@implementation Student

-(instancetype)initWithAge:(int)age
{
    if (self = [super init]) {
        _age = age;
    }
    return [self autorelease];
}

+(instancetype)studentWithAge:(int)age
{
    return [[self alloc] initWithAge:age];
}

-(void)dealloc
{
    [super dealloc];
    NSLog(@"%@ dealloc...", self);
}

@end

File: Day34_Swift基础/09_Swift面向对象相关/01_Swift面向对象相关/Student.swift
//
//  Student.swift
//  01_Swift面向对象相关
//
//  Created by Chiang on 16/4/9.
//  Copyright © 2016年 Google. All rights reserved.
//

import UIKit

class Student: Person {
    override init() {
        super.init()
        name = "YiJianLian"
        age = 27
    }
    
    init(name:String, age:Int) {
        // 参数名和属性名相同时，属性名需要加上self来区分。
        super.init()
        self.name = name
        self.age = age
    }
}
